WordPress/Sniffs/NamingConventions/ValidFunctionNameSniff.php

```diff
@@ -133,7 +133,50 @@ protected function processTokenWithinScope(PHP_CodeSniffer_File $phpcsFile, $sta
 
     }//end processTokenWithinScope()
 
+	/**
+	 * Returns the name of the class that the specified class implements.
+	 *
+	 * Returns FALSE on error or if there is no implemented class name.
+	 *
+	 * @param int $stackPtr The stack position of the class.
+	 * @param PHP_CodeSniffer_File $phpcsFile The stack position of the class.
+	 *
+	 * @see PEAR_Sniffs_NamingConventions_ValidFunctionNameSniff::findExtendedClassName()
+	 *
+	 * @todo This needs to be upstreamed and made part of PHP_CodeSniffer_File.
+	 *
+	 * @return string
+	 */
+	public function findImplementedClassName( $stackPtr, $phpcsFile ) {
+        $tokens = $phpcsFile->getTokens();
+
+		// Check for the existence of the token.
+		if ( isset( $tokens[ $stackPtr ] ) === false ) {
+			return false;
+		}
+		if ( $tokens[ $stackPtr ]['code'] !== T_CLASS ) {
+			return false;
+		}
+		if ( isset( $tokens[ $stackPtr ]['scope_closer'] ) === false ) {
+			return false;
+		}
+		$classCloserIndex = $tokens[ $stackPtr ]['scope_closer'];
```

Comment:
@JDGrimes it's because it is being passed as the third argument (`$end`) to `PHP_CodeSniffer_File::findNext( $types, $start, $end, $exclude, $value, $local )`, meaning it should stop looking once it gets to that point. Right?